src/client/QXmppPubSubManager.cpp

```diff
@@ -160,6 +163,40 @@ QXmppPubSubManager::~QXmppPubSubManager()
 {
 }
 
+///
+/// Requests all features of a pubsub service via service discovery.
+///
+/// This uses a \xep{0030, Service Discovery} info request to get a list of
+/// features.
+///
+/// \param serviceJid JID of the entity hosting the pubsub service
+/// \param onlyForPepService whether to retrieve features only if the pubsub
+///        service is a PEP service
+/// \return
+///
+QFuture<QXmppPubSubManager::FeaturesResult> QXmppPubSubManager::requestFeatures(const QString &serviceJid, const bool onlyForPepService)
```

Comment:
The documentation is missing:
 * the function checks that the service has the correct identity
 * what happens / is returned when the service doesn't have the correct identity